pkg/yttlibrary/assert.go

```diff
@@ -8,26 +8,51 @@ import (
 
 	"github.com/k14s/starlark-go/starlark"
 	"github.com/k14s/starlark-go/starlarkstruct"
+	"github.com/k14s/starlark-go/syntax"
+	"github.com/vmware-tanzu/carvel-ytt/pkg/experiments"
+	"github.com/vmware-tanzu/carvel-ytt/pkg/orderedmap"
+	"github.com/vmware-tanzu/carvel-ytt/pkg/spell"
 	"github.com/vmware-tanzu/carvel-ytt/pkg/template/core"
 )
 
-var (
-	AssertAPI = starlark.StringDict{
+// NewAssertModule constructs a new instance of AssertModule, respecting the "validations" experiment flag.
+func NewAssertModule() AssertModule {
+	return AssertModule{}
+}
+
+// AssertModule contains the definition of the @ytt:assert module.
+type AssertModule struct{}
+
+// AsModule produces the corresponding Starlark module definition suitable for use in running a Starlark program.
+func (m AssertModule) AsModule() starlark.StringDict {
+	members := starlark.StringDict{
+		"equals": starlark.NewBuiltin("assert.equals", core.ErrWrapper(m.Equals)),
+		"fail":   starlark.NewBuiltin("assert.fail", core.ErrWrapper(m.Fail)),
+		"try_to": starlark.NewBuiltin("assert.try_to", core.ErrWrapper(m.TryTo)),
+	}
+	if experiments.IsValidationsEnabled() {
+		members = starlark.StringDict{
+			"equals":       starlark.NewBuiltin("assert.equals", core.ErrWrapper(m.Equals)),
+			"fail":         starlark.NewBuiltin("assert.fail", core.ErrWrapper(m.Fail)),
+			"try_to":       starlark.NewBuiltin("assert.try_to", core.ErrWrapper(m.TryTo)),
```

Comment:
instead of overriding members entirely, we should just be able to set new keys (keeping baseline the same)